Lib/test/test_tokenize.py

```diff
@@ -1630,13 +1630,34 @@ def test_pathological_trailing_whitespace(self):
     def test_comment_at_the_end_of_the_source_without_newline(self):
         # See http://bugs.python.org/issue44667
         source = 'b = 1\n\n#test'
-        expected_tokens = [token.NAME, token.EQUAL, token.NUMBER, token.NEWLINE, token.NL, token.COMMENT]
+        expected_tokens = [
+            TokenInfo(type=token.ENCODING, string='utf-8', start=(0, 0), end=(0, 0), line=''),
+            TokenInfo(type=token.NAME, string='b', start=(1, 0), end=(1, 1), line='b = 1\n'),
+            TokenInfo(type=token.OP, string='=', start=(1, 2), end=(1, 3), line='b = 1\n'),
+            TokenInfo(type=token.NUMBER, string='1', start=(1, 4), end=(1, 5), line='b = 1\n'),
```

Comment:
Shouldn't the next token start where the previous one ends?
```suggestion
            TokenInfo(type=token.NAME, string='b', start=(1, 0), end=(1, 1), line='b = 1\n'),
            TokenInfo(type=token.OP, string='=', start=(1, 1), end=(1, 2), line='b = 1\n'),
            TokenInfo(type=token.NUMBER, string='1', start=(1, 2), end=(1, 3), line='b = 1\n'),
```

Reply:
Not because there are spaces in the middle and those are not included in the string.

This is also the output of the old tokenizer. Consider this:

```
import io, pprint, tokenize
code = 'b = 1\n\n#test'

for token in tokenize.generate_tokens(io.StringIO(code).readline):
    pprint.pprint(token)
```

This prints in 3.11:

```
TokenInfo(type=1 (NAME), string='b', start=(1, 0), end=(1, 1), line='b = 1\n')
TokenInfo(type=54 (OP), string='=', start=(1, 2), end=(1, 3), line='b = 1\n')
TokenInfo(type=2 (NUMBER), string='1', start=(1, 4), end=(1, 5), line='b = 1\n')
TokenInfo(type=4 (NEWLINE), string='\n', start=(1, 5), end=(1, 6), line='b = 1\n')
TokenInfo(type=62 (NL), string='\n', start=(2, 0), end=(2, 1), line='\n')
TokenInfo(type=61 (COMMENT), string='#test', start=(3, 0), end=(3, 5), line='#test')
TokenInfo(type=62 (NL), string='', start=(3, 5), end=(3, 5), line='#test')
TokenInfo(type=0 (ENDMARKER), string='', start=(4, 0), end=(4, 0), line='')
```